### PR TITLE
internal/storetesting: more stringent AssertJSONCall test

### PR DESCRIPTION
**Pull request by rogpeppe**
This makes sure that _all_ the data returned is as expected.

### CODE REVIEW

**Comment by bac** [comment]
I think this is a much cleaner solution and it has the added benefit of working properly.  :+1:

File: internal/storetesting/http.go [hunk 1]
```diff
@@ -33,10 +32,21 @@ func AssertJSONCall(
 		c.Assert(rec.Body.Bytes(), gc.HasLen, 0)
 		return
 	}
-	resp := reflect.New(reflect.TypeOf(expectBody))
-	err := json.Unmarshal(rec.Body.Bytes(), resp.Interface())
+	// Rather than unmarshaling into something of the expected
```
Comment (bac) [inline internal/storetesting/http.go:35]: typo: unmarshalling

**Comment by frankban** [comment]
:+1:

File: internal/storetesting/http.go (hunk 1 above)
Comment (rogpeppe) [inline internal/storetesting/http.go:35]: leaving as is, as discussed.

**Comment by rogpeppe** [comment]
$$shipit$$

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/charmstore-merge/18

**Pull request merged**